os/registry-authentication.md

```diff
@@ -0,0 +1,234 @@
+
+```
+$ kubectl get my-favorite-registry-secret
+NAME                            TYPE                      DATA      AGE
+my-favorite-registry-secret     kubernetes.io/dockercfg   1         30m
+```
+
+The secret can be used in a Pod spec with the `imagePullSecrets` variable:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: somepod
+  namespace: all
+spec:
+  containers:
+    - name: web
+      image: registry.example.io/v0/giffeeLover93/somerepo
+  
+  imagePullSecrets:
+    - name: my-favorite-registry-secret
+```
+
+For more information read the [docker-registry Kubernetes secret][k8s-docker-registry] and [Kubernetes imagePullSecrets][k8s-image-pull] documents on kubernetes.io.
+
+### rkt
+
+rkt stores registry-authentication in a JSON file stored in the directory `/etc/rkt/auth.d/`.
+
+#### /etc/rkt/auth.d/registry.example.io.json
+
+An example of this configuration is:
+
+```json
+{
+  "rktKind": "auth",
+  "rktVersion": "v1",
+  "domains": [
+    "https://registry.example.io/v0/"
+  ],
+  "type": "basic",
+  "credentials": {
+    "user": "giffeeLover93",
+    "password": "passphrases are great!"
+  }
+}
+```
+
+**Note**: While you *can* embed your password in plaintext in this file, you should try using a disposable token instead! Check your registry documentation to see if it offers token-based authentication.
+
+Now rkt will authenticate with `https://registry.example.io/v0/` using the provided credentials when an image is pulled from there.
+
+For more information about rkt credentials read the [rkt configuration docs on coreos.com][rkt-config].
+
+Just like with the Docker config, this file can be copied to `/etc/rkt/auth.d/registry.example.io.json` on a CoreOS node during first-boot using [Ignition][ignition].
+
+### Mesos
+
+Mesos uses gzipped compressed archive of a `.docker/config.json` (directory and file) to access private repositories.
+
+Once you have followed the above steps to [create the docker registry auth config file][docker-instructions] create your Mesos configuration using `tar`:
+
+```
+$ tar cxf ~/.docker/config.json
+```
+
+The archive secret is referenced via the `uris` field in a container specification file:
+
+```json
+{
+  "id": "/some/name/or/id",
+  "cpus": 1,
+  "mem": 1024,
+  "instances": 1,
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "https://registry.example.io/v0/giffeeLover93/some-image",
+      "network": "HOST"
+    }
+  },
+  
+  "uris":  [
+      "file:///path/to/registry.example.io.tar.gz"
+  ]
+}
+```
+
+More thorough information about configuring Mesos registry authentication can be found on the ['Using a Private Docker Registry'][mesos-registry] Mesos/Marathon documentation.
+
+## Copying the config file with ignition
+
+[Ignition][ignition] is a low level system configuration utility used to provision a CoreOS node on it's first boot. Here is an example of copying one of the configuration files mentioned above onto the appropriate path in the node:
```

Comment:
I feel like we're missing what this gets you. "Doing this on first boot allows you machine to immediately have access to all your private images blah blah"